src/Bundle/Form/DataTransformer/RecursiveTransformer.php

```diff
@@ -28,7 +29,7 @@ public function __construct(DataTransformerInterface $decoratedTransformer)
     }
 
     /** @param Collection|null $value */
-    public function transform($value): Collection
+    public function transform($value): ReadableCollection
```

Comment:
I had mixed feelings about whether we're not breaking anything here - in the end, it's a change of the type of returned value. But because it's a final class, used only inside forms processing and in fact, the BC is not broken, as we could just return _more_ not _less_... It's safe to have it merged 🖖

Reply:
Yes, I've checked it (cf our slack), there's no BC break